vir/src/context.rs

```diff
@@ -515,14 +552,18 @@ impl<'tcx> VirCtxt<'tcx> {
     }
 
     pub fn mk_conj<'vir>(&'vir self, elems: &[Expr<'vir>]) -> Expr<'vir> {
-        if elems.len() == 0 {
-            return self.mk_bool::<true>();
-        }
-        let mut e = elems[0];
-        for i in 1..elems.len() {
-            e = self.mk_bin_op_expr(BinOpKind::And, e, elems[i]);
-        }
-        e
+        elems.split_first().map(|(first, rest)| {
+            rest.iter().fold(*first, |acc, e| {
```

Comment:
Maybe this should rather be an `rfold`? To produce `a1 && (a2 && (a3 && a4))` rather than `((a1 && a2) && a3) && a4`. (Marginally more readable.)